DataManagementSystem/DB/FileCatalogDB.py

```diff
@@ -1021,14 +1021,21 @@ def rebuildDirectoryUsage(self):
     result = self.dtree._rebuildDirectoryUsage()
     return result
 
-  def repairCatalog(self, directoryFlag=True, credDict={}):
+  def repairCatalog(self, credDict={}):
     """ Repair catalog inconsistencies
     """
-    result = S_OK()
-    if directoryFlag:
-      result = self.dtree.recoverOrphanDirectories(credDict)
 
-    return result
+    result = self.securityManager.hasAdminAccess(credDict)
+    if not result['OK']:
+      return result
+    if not result['Value']:
```

Comment:
How about returning `EACCES`  ? (purist would probably argue for `EPERM`, but since we use `EACCES` everywhere, better stick to that one and only one)

Reply:
Done